File: lambda-cors-fix.js
// Add this to the beginning of your Lambda handler response:

return {
    statusCode: 200,
    headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Headers': 'content-type',
        'Access-Control-Allow-Methods': 'POST,OPTIONS'
    },
    body: JSON.stringify({
        message: "Successfully generated and merged audio!",
        audio_url: audioUrl
    })
};

File: lambda_cors_fix.py
import json

def lambda_handler(event, context):
    # Your existing logic here
    
    # Return response with CORS headers
    return {
        'statusCode': 200,
        'headers': {
            'Access-Control-Allow-Origin': '*',  # or 'http://127.0.0.1:8080'
            'Access-Control-Allow-Headers': 'Content-Type,X-Requested-With',
            'Access-Control-Allow-Methods': 'POST,OPTIONS'
        },
        'body': json.dumps({
            'audio_url': 'your_audio_url_here'
        })
    }